### PR TITLE
Enforce tensorboard minimum version as 1.15

### LINKED ISSUE #34028: 'LocalFileSystem' object has no attribute 'makedirs'
## To Reproduce

While working with the `add_embedding` method for TensorBoard (for the first time), there seems to be an issue with one of the lines in the `writer.py` files that is invoked. The simplest way to reproduce would be to test the **Adding a "Projector" to TensorBoard** section of the TensorBoard tutorial (Link: https://pytorch.org/tutorials/intermediate/tensorboard_tutorial.html#adding-a-projector-to-tensorboard)

```
warning: Embedding dir exists, did you set global_step for add_embedding()?
  File "training_visualization.py", line 87, in <module>
    writer.add_embedding(features, metadata=class_labels, label_img=images.unsqueeze(1))
  File "/home/nivii/Desktop/python_environments/pytorch_py36/lib/python3.6/site-packages/torch/utils/tensorboard/writer.py", line 794, in add_embedding
    make_tsv(metadata, save_path, metadata_header=metadata_header)
  File "/home/nivii/Desktop/python_environments/pytorch_py36/lib/python3.6/site-packages/torch/utils/tensorboard/_embedding.py", line 20, in make_tsv
    fs.write(fs.join(save_path, 'metadata.tsv'), metadata_bytes, binary_mode=True)
AttributeError: 'LocalFileSystem' object has no attribute 'write'


```

## Expected behavior

No error, I presume. I am not sure where is should be setting the `global_step` in the case of the tutorial, while following it sequentially.

## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

 - PyTorch Version (e.g., 1.0): 1.4.0
 - OS (e.g., Linux): Linux
 - How you installed PyTorch (`conda`, `pip`, source): `pip`
 - Build command you used (if compiling from source):
 - Python version: 3.6
 - CUDA/cuDNN version: 10.0/7/5/0
 - GPU models and configuration: NVIDIA RTX2060
 - Any other relevant information: Tensorboard 1.14.0

## Additional context

cc @ezyang @gchanan @zou3519

### PR DESCRIPTION
**Pull request by peterbell10**
Fixes #34028

### CODE REVIEW

**Review by ezyang — APPROVED** [review]

**Comment by ezyang** [comment]
Unfortunately I can't land this right now because our internal version of tensorboard is too old (lol).

File: torch/utils/tensorboard/__init__.py [hunk 1]
```diff
@@ -1,6 +1,9 @@
-try:
-    from tensorboard.summary.writer.record_writer import RecordWriter  # noqa F401
-except ImportError:
-    raise ImportError('TensorBoard logging requires TensorBoard with Python summary writer installed. '
-                      'This should be available in 1.14 or above.')
+import tensorboard
+from distutils.version import LooseVersion
+if LooseVersion(tensorboard.__version__) < LooseVersion('1.15'):
+    raise ImportError('TensorBoard logging requires TensorBoard version 1.15 or above')
```
Comment (lanpa) [inline torch/utils/tensorboard/__init__.py:4]: If TensorBoard is not installed, will this error message show?

**Review by lanpa — COMMENTED** [review]
If TensorBoard is not installed, will the error message be meaningful?

File: torch/utils/tensorboard/__init__.py (hunk 1 above)
Comment (peterbell10) [reply torch/utils/tensorboard/__init__.py:4]: `import tensorboard` would fail with a normal `ImportError`.

**Comment by ezyang** [comment]
@orionr, could you figure out who the right person to talk to about the internal version of TensorBoard is here?

**Comment by orionr** [comment]
@sanekmelnikov looks like we might need to upgrade our internal TensorBoard version.  Can you investigate? Thanks.

**Comment by orionr** [comment]
Actually, we have 2.0 internally... let me investigate with @sanekmelnikov

**Pull request closed without merging**